apps/api/src/app/activity/usecases/get-workflow-run/get-workflow-run.usecase.ts

```diff
@@ -245,7 +248,13 @@ export class GetWorkflowRun {
       environmentId: workflowRun.environment_id,
       internalSubscriberId: workflowRun.subscriber_id,
       subscriberId: workflowRun.external_subscriber_id || undefined,
-      status: mapWorkflowRunStatusToDto(workflowRun.status, stepRuns),
+      status: mapWorkflowRunStatusToDto(workflowRun.status),
+      deliveryLifecycleStatus: (await this.workflowRunService.getDeliveryLifecycle({
+        notificationId: workflowRun.workflow_run_id,
+        environmentId: workflowRun.environment_id,
+        organizationId: workflowRun.organization_id,
+        subscriberId: workflowRun.subscriber_id,
+      })).deliveryLifecycleStatus as unknown as WorkflowRunDeliveryLifecycleStatusDtoEnum,
```

Comment:
Is this calculated in runtime here? Why not just get it from the workflow run entity?

Reply:
we can bug then we will not support the data backward, if we are okay with that i can update this code.
but i would suggest keeping this one for now, so we will have the data from the past workflow runs for a couple of weeks

Reply:
done